client/components/AddTestPlanToQueue/AddTestPlanToQueueModal.jsx

```diff
@@ -0,0 +1,273 @@
+import React, { useState, useEffect } from 'react';
+import { Button, Modal, Form } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { useQuery, useMutation } from '@apollo/client';
+import {
+    ADD_TEST_QUEUE_MUTATION,
+    POPULATE_ADD_TEST_PLAN_TO_QUEUE_MODAL_QUERY
+} from '../TestQueue/queries';
+
+const AddTestPlanToQueueModal = ({
+    show = false,
+    handleClose = () => {},
+    handleAddToTestQueue = () => {}
+}) => {
+    const [ats, setAts] = useState([]);
+    const [browsers, setBrowsers] = useState([]);
+    const [testPlans, setTestPlans] = useState([]);
+    const [filteredTestPlans, setFilteredTestPlans] = useState([]);
+    const [testPlanVersions, setTestPlanVersions] = useState([]);
+    const [selectedAt, setSelectedAt] = useState('');
+    const [selectedBrowser, setSelectedBrowser] = useState('');
+    const [atVersion, setAtVersion] = useState('');
+    const [browserVersion, setBrowserVersion] = useState('');
+    const [selectedTestPlan, setSelectedTestPlan] = useState('');
+    const [selectedTestPlanVersion, setSelectedTestPlanVersion] = useState('');
+
+    // eslint-disable-next-line no-unused-vars
+    const { loading, error, data } = useQuery(
+        POPULATE_ADD_TEST_PLAN_TO_QUEUE_MODAL_QUERY
+    );
+    const [addTestPlanReport] = useMutation(ADD_TEST_QUEUE_MUTATION);
+
+    useEffect(() => {
+        if (data) {
+            const {
+                ats = [],
+                browsers = [],
+                testPlanVersions: testPlans = []
```

Comment:
Think I was actually just avoiding a variable clash here and couldn't come up with something better at the time 🙂. Apologies on that confusion.

>  I think the right way to fix it would be to update the query to something like this:

I do know the query you speak of though. When we initially took a look at it, it doesn't return what was exactly wanted here and I do know you had further plans there so didn't want to have a clash in that method. However, there is a `filteredTestPlans` variable in this file which I believe may come close to what we'd want in that GQL resolver so having that functionality expanded and completed on the resolver would be great.